### PR TITLE
Add support for editing keyboard information in package editor, plus basic BCP-47 support to make this work

### PR DESCRIPTION
**Pull request by mcdurdin**
This PR adds support for editing the `<Keyboards/>` section of .kps files and building them into the package. It also puts in some of the framework for BCP 47 support in order for the package editor to generate clean BCP 47 codes. Most of the BCP 47 support is left for a future PR - some comments are in place with `// TODO: BCP47:` to prompt us when we get to that.

You'll see that I've started using namespacing for Keyman units. Namespacing was introduced to Delphi a few years ago. The namespacing rules I am adopting are:

1. `Keyman.` is the base namespace for all Keyman code.
2. If a unit is specific to a project I then append its namespace, e.g. `Keyman.Developer.`
3. UI units (e.g. imports `VCL` units or has any other UI) are then `Keyman.[Developer.]UI.`
4. Console units (e.g. uses `writeln`) are `Keyman.[Developer.]Console.`
5. System units are those that have no console or UI I/O, and are `Keyman.[Developer.]System.`
6. It is good to append another namespace if it fits into a sensible space, e.g. `Keyman.System.Standards.`. Use common sense here.
7. Finally, note that base unit names (stripping off the namespace) should still be unique so don't use generic names. i.e. use `Keyman.System.KeyboardUtils` or `Keyman.System.Keyboards.KeyboardUtils` not `Keyman.System.Keyboard.Utils`.

Once this PR is approved, I will aim to push those rules into the wiki. We'll move units over to the new namespaces over time, but there's not a lot of value in doing that in a huge unit of work I think.

### CODE REVIEW

File: windows/src/global/delphi/keyboards/Keyman.System.KeyboardJSInfo.pas [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  System.SysUtils;
+
+type
+  TKeyboardJSInfo = class
+  private
+    FVersion: string;
+    FName: string;
+    FMnemonic: Boolean;
+    FRTL: Boolean;
+
```
Comment (darcywong00) [inline windows/src/global/delphi/keyboards/Keyman.System.KeyboardJSInfo.pas]: What about parsing for `this.KI` and including `keyboardID` in this struct?

File: windows/src/test/inffiletests/packageinf.pas [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+begin
+  CoInitializeEx(nil, COINIT_APARTMENTTHREADED);
+  with TKMPInfFile.Create do
+  try
+    FileName := 'c:\temp\kmp.inf';
+    LoadIni;
+    FileName := 'c:\temp\kmp.xml';
+    SaveXML;
+    FileName := 'c:\temp\kmp.json';
+    SaveJSON;
```
Comment (darcywong00) [inline windows/src/test/inffiletests/packageinf.pas]: Is `c:\temp\` guaranteed to exist? What about using `System.IOUtils.TPath.GetTempPath()`?

File: windows/src/global/delphi/keyboards/Keyman.System.KeyboardJSInfo.pas (hunk 1 above)
Comment (mcdurdin) [reply windows/src/global/delphi/keyboards/Keyman.System.KeyboardJSInfo.pas]: `keyboardID` is already captured from the name of the file. We can do this if we need it at a future date.

File: windows/src/test/inffiletests/packageinf.pas (hunk 2 above)
Comment (mcdurdin) [reply windows/src/test/inffiletests/packageinf.pas]: This is a test app for experimenting with inf files. I saved it just for posterity; we don't really need it (perhaps I could take it out of the repo). In any case, we don't really need to spend time on it.

**Pull request merged**